devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusTestConfig.java

```diff
@@ -34,7 +33,7 @@ public void setupTest() {
 
             for (Test test : getProject().getTasks().withType(Test.class)) {
                 final Map<String, Object> props = test.getSystemProperties();
-                props.put(BootstrapClassLoaderFactory.PROP_DEPLOYMENT_CP, deploymentCp);
+                //props.put(BootstrapClassLoaderFactory.PROP_DEPLOYMENT_CP, deploymentCp);
```

Comment:
That's an interesting one. If it's not needed any more, this whole task is questionable. We shouldn't be resolving the app model here, at a minimum.

Reply:
IIUIC, here is a potential issue with this change. Gradle and Maven perform version convergence differently. The way it currently is in master we are using the same Gradle resolver for tests and actual builds.
With this change it looks like we are going to be using the Maven resolver for the tests and the Gradle one for the actual builds. And the dev mode will always be using the Maven resolver in `DevModeMain`.

Reply:
This is something I am not sure about, and wanted some feedback on.

If this is removed we fall back to class path based discovery in the maven resolver (unless there happens to be a pom present as well). The main advantage of this is that it should allow gradle tests to also run in the IDE, which does not work at the moment. 

If that is not possible then we need to somehow encode the app model into this system prop and restore the old behaviour.

Reply:
While it does not look elegant in this case, we are ending up testing against a different set of deps than the ones used for the build. They should be aligned. Either we use those resolved by Gradle all the way or we use those that we resolve in QuarkusBootstrap.

Reply:
By not looking elegant I refer to the classpath shared through the system property.